autoload/ale/virtualtext.vim

```diff
@@ -245,3 +253,37 @@ function! ale#virtualtext#SetTexts(buffer, loclist) abort
         endif
     endfor
 endfunction
+
+function! ale#virtualtext#IsSupported() abort
+    return s:has_virt_text
+endfunction
+
+function! ale#virtualtext#IsEmulated() abort
+    return s:emulate_virt
+endfunction
+
+function! ale#virtualtext#CountTexts(buffer) abort
```

Comment:
Last move this code to the test file instead, as it's just for tests.

Reply:
OK, so this is my first foray into Vimscript, so it is very possible that I'm missing something.  But all three of these functions access script-local variables, so I don't know of a good way to move them into another file.  If there is a clean way to handle this that is being used in other tests, could you point me to an example I could emulate?

Reply:
I'll see myself if I can eliminate the need for these being in the file here. I don't like to overburdern the scripts with things just for tests, but sometimes it's the only way. I'm pretty sure we can rewrite the same conditions as above into tests or simplify them in tests with version checks, given that we only test against certain versions in Docker anyway, and we can probably just get all available marks instead of specific ones in tests. Something like that.

Reply:
OK, so I just pushed up a new version with an alternate approach:  now the functions are only defined if the `g:ale_private_testing_mode` variable is set to true, and that variable is only defined in `test/vimrc`.  That way the functions don't have to be modified, but the global namespace is kept clean when not testing.  Thoughts?